src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -315,4 +328,51 @@ private void saveWithDifferentEncoding(Path file, boolean selectedOnly, Charset
             }
         }
     }
+
+    /**
+     * Handle JabRef git integration action. This method is called when the user save the database.
+     */
+    public void automaticGitUpdate(Path filePath) throws IOException, GitAPIException {
+        GitHandler git = new GitHandler(filePath.getParent(), preferences.getGitPreferences());
+        List<GitChange> changes = new ArrayList<GitChange>();
+        BibGitContext bibGitContext = new BibGitContext();
+        if (preferences.getGitPreferences().getAutoCommit()) {
+            String automaticCommitMsg = "Automatic update via JabRef";
+            git.createCommitWithSingleFileOnCurrentBranch(filePath.getFileName().toString(), automaticCommitMsg);
+        }
+        if (preferences.getGitPreferences().getAutoSync()) {
+            try {
+                git.pullOnCurrentBranch();
+            } catch (IOException ex1) {
+                if (ex1.getMessage().equals("No git credentials")) {
+                    GitCredentialsDialogView gitCredentialsDialogView = new GitCredentialsDialogView();
+
+                    gitCredentialsDialogView.showGitCredentialsDialog();
+
+                    UsernamePasswordCredentialsProvider credentialsProvider = new UsernamePasswordCredentialsProvider(
+                            gitCredentialsDialogView.getGitUsername(),
+                            gitCredentialsDialogView.getGitPassword()
+                    );
+
+                    git.setCredentialsProvider(credentialsProvider);
+                    try {
+                        git.pullOnCurrentBranch();
+                    } catch (IOException ex2) {
+                        if (ex2.getMessage().equals("HEAD is detached")) {
+                            GitChangeResolverDialog GitChangeResolverDialog = new GitChangeResolverDialog(changes, bibGitContext, "Merge issues");
+                        } else {
+                            throw new RuntimeException(ex2.getMessage());
```

Comment:
NEVER EVER a RuntimeException. This tears down whole JabRef.

A user NEVER wants that JabRef crashes ON SAVE.